pysb/kappa.py

```diff
@@ -30,6 +29,99 @@
 except ImportError:
     pass
 
+# Cached value of Kappa program paths
+_kasim_path = None
+_kasa_path = None
+
+def _check_dist_dir(kappa_prog, dist_dir):
+    kappa_prog_path = os.path.join(dist_dir, kappa_prog)
+    if os.access(kappa_prog_path, os.F_OK):
+        return kappa_prog_path
+    else:
+        return None
+
+def set_kappa_path(dir):
+    """Set the path to the KaSim and KaSa executables.
+
+    Parameters
+    ----------
+    dir : string
+        Directory containing KaSim and Kasa executables.
+    """
+
+    global _kasim_path
+    global _kasa_path
+    # Check locations of both KaSim and KaSa
+    kasim_path = _check_dist_dir('KaSim', dir)
+    kasa_path = _check_dist_dir('KaSa', dir)
+
+    # Check that KaSim executable exists and is not a directory
+    if kasim_path is None:
+        raise Exception('Could not find KaSim in %s.' % os.path.abspath(dir))
+    else:
+        _kasim_path = kasim_path
+
+    # Check KaSa, but only raise a warning if it's not found because it's
+    # not installed by default with the KaSim distribution.
+    if kasa_path is None:
+        warnings.warn('Could not find KaSa in %s; install KaSa for static '
+                      'analysis features.' % os.path.abspath(dir))
+    else:
+        _kasa_path = kasa_path
+
+def _get_kappa_path(kappa_prog):
+    """
+    Return the path to the KaSim or KaSa executable.
+
+    Looks for a KaSim/KaSa distribution at the path stored in the KAPPAPATH
+    environment variable if that's set, or else in a few hard-coded standard
+    locations.
+    """
+
+    if kappa_prog not in ['KaSim', 'KaSa']:
+        raise ValueError("kappa_prog must be one of 'KaSim', 'KaSa'.")
+
+    global _kasim_path
+    global _kasa_path
+
+    # Just return cached value if it's available
+    if kappa_prog == 'KaSim' and _kasim_path is not None:
+        return _kasim_path
+    elif kappa_prog == 'KaSa' and _kasa_path is not None:
+        return _kasa_path
+
+    path_var = 'KAPPAPATH'
+    dist_dirs = [
+        '/usr/local/share/KaSim',
+        'c:/Program Files/KaSim',
+        ]
+
+    # First check the environment variable, which has the highest precedence
+    if path_var in os.environ:
+        kappa_prog_path = _check_dist_dir(kappa_prog, os.environ[path_var])
+        if not kappa_prog_path:
+            raise Exception('Environment variable %s is set to but %s could'
```

Comment:
Looks like you were going to include the value of `$KAPPAPATH` after 'to' here but forgot to finish.